core/src/main/java/bisq/core/xmr/knaccc/monero/address/WalletAddress.java

```diff
@@ -121,7 +123,21 @@ public static String getSubaddressBase58(Scalar privateViewKey,
 
     }
 
-    public String getSubaddressBase58(String privateViewKeyHex, long accountId, long subaddressId) {
+    public String getSubaddressBase58(String privateViewKeyHex, long accountId, long subaddressId) throws InvalidWalletAddressException {
+        if (!checkPrivateViewKey(privateViewKeyHex)) {
+            throw new InvalidWalletAddressException("Wrong private view key for main address");
+        }
         return getSubaddressBase58(new Scalar(privateViewKeyHex), hexToBytes(getPublicSpendKeyHex()), accountId, subaddressId);
     }
+
+    public boolean checkPrivateViewKey(String privateViewKey) {
+        return arePubPrivKeysRelated(this.publicViewKeyHex, privateViewKey);
+    }
+
+    public static boolean arePubPrivKeysRelated(String publicKey, String privateKey) {
+        Scalar m = new Scalar(privateKey);
```

Comment:
Why is the var called m?

Reply:
according to those docs, and the code in this module, m is the "per sub-address scalar", a would be the private view key and A would be the public view key but this doesn't really matter that much